src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,164 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double STEP_UPHILL = 1.5d;
+    private static final int BASE_INDEX_FROM = 0;
+    private static final Object[] EMPTY_ARRAY = new Object[0];
+    private Object[] dataCollection;
+    private int size;
+
+    public ArrayList() {
+        this.dataCollection = EMPTY_ARRAY;
+    }
+
+    public ArrayList(int number) {
+        if (number > 0) {
+            this.dataCollection = new Object[number];
+        } else {
+            if (number != 0) {
+                throw new IllegalArgumentException("Illegal argument: " + number);
+            } else {
+                this.dataCollection = EMPTY_ARRAY;
+            }
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        if (this.size == this.dataCollection.length) {
+            this.dataCollection = grow(this.size + 1);
+        }
+        this.dataCollection[this.size] = value;
+        this.size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > this.size || index < 0) {
+            sendException(index);
+        } else {
+            if (index == this.size) {
+                add(value);
+            } else {
+                int afterIndex = this.size - index;
+                Object[] afterArray;
+                afterArray = copyOf(this.dataCollection, afterIndex, index);
+                this.dataCollection = grow(this.size + 1);
+                this.dataCollection[index] = value;
+                for (Object o : afterArray) {
+                    index++;
+                    this.dataCollection[index] = o;
+                }
+                this.size++;
```

Comment:
make it like this 
![image](https://user-images.githubusercontent.com/20326858/224301178-95a75128-a7dc-49f6-a7ca-2b0d9d8b8b2d.png)